lib/src/main/java/com/diffplug/spotless/kotlin/KtLintStep.java

```diff
@@ -110,7 +110,8 @@ static final class State implements Serializable {
 			this.version = version;
 			this.userData = new TreeMap<>(userData);
 			this.editorConfigOverride = new TreeMap<>(editorConfigOverride);
-			this.jarState = JarState.from(MAVEN_COORDINATE + version, provisioner);
+			this.jarState = JarState.from((version.startsWith("0.") ? MAVEN_COORDINATE_0_DOT : MAVEN_COORDINATE_1_DOT) + version,
+					provisioner);
```

Comment:
> io.github.oshai:kotlin-logging-jvm:5.1.0 is added in classpath, but can't be found?

Are you sure? Maybe try looking at what gets resolved right here? Could also add `kotlin-logging-jvm` here if needed, though that would be ugly...

Reply:
Some sources jars were resolved:
```
...
Resolved: kotlin-logging-jvm-5.1.0-sources.jar
Resolved: kotlin-compiler-embeddable-1.9.10.jar
Resolved: ec4j-core-0.3.0.jar
Resolved: logback-core-1.3.5.jar
Resolved: slf4j-api-2.0.4.jar
Resolved: poko-annotations-jvm-0.15.0-sources.jar
Resolved: kotlin-stdlib-jdk7-1.9.10.jar
Resolved: kotlin-stdlib-1.9.10.jar
Resolved: kotlin-script-runtime-1.9.10.jar
Resolved: kotlin-reflect-1.6.10.jar
Resolved: kotlin-daemon-embeddable-1.9.10.jar
Resolved: trove4j-1.0.20200330.jar
Resolved: kotlin-stdlib-common-1.9.10.jar
Resolved: annotations-13.0.jar
Resolved: sarif4k-jvm-0.4.0-sources.jar
...
```

Reply:
Straaaaange. The [pom file](https://repo1.maven.org/maven2/com/pinterest/ktlint/ktlint-cli/1.0.0/ktlint-cli-1.0.0.pom) seems normal, and the [module file](https://repo1.maven.org/maven2/com/pinterest/ktlint/ktlint-cli/1.0.0/ktlint-cli-1.0.0.module) seems normal enough too. I didn't trace it out through the transitives though, I wonder if one of them has its transtives specified in a strange way. We can look into it, or we can wait for it to get resolved in a `1.0.1`.

Reply:
Seems they are both Kotlin multiplatform libs:

```
Resolved: kotlin-logging-jvm-5.1.0-sources.jar
Resolved: poko-annotations-jvm-0.15.0-sources.jar
Resolved: sarif4k-jvm-0.4.0-sources.jar
```

Reply:
And if I'm using `compileOnly` here, it works, see https://github.com/Goooler/spotless/pull/2#issuecomment-1713136961.

Reply:
The root cause is https://github.com/diffplug/spotless/blob/078637874168f82f6455354dfc797568d821b9c0/plugin-gradle/src/main/java/com/diffplug/gradle/spotless/GradleProvisioner.java#L123-L125

If we remove `BUNDLING_ATTRIBUTE`, the resolved files are normal. But seems we can't remove it due to #1013.

Reply:
https://github.com/diffplug/spotless/blob/078637874168f82f6455354dfc797568d821b9c0/lib/build.gradle#L44-L47


I think we should special-case the `Bundling.EXTERNAL` thing to the older `KtLint` versions. If KtLint 1.0 really keeps a stable API for a whole year, then we could remove the old shims and their related bundling workarounds.

Reply:
Unfortunately, it's still necessary in Ktlint 1.0.

Reply:
Hm, I'm confused because this seems to not match this one

> If we remove BUNDLING_ATTRIBUTE, the resolved files are normal.

Feel free to special-case `GradleProvisioner` if it helps with anything.

Reply:
919b4811448e1ec11ceadfebc8bcfc8f48137ad4